src/audio.cpp

```diff
@@ -0,0 +1,82 @@
+#include "audio.h"
+
+#include "util/log.h"
+#include "math/smek_math.h"
+
+namespace Audio {
+
+void audio_callback(AudioStruct *audio_struct, u8 *stream, int len) {
```

Comment:
If the first thing you do is cast it to a float, and calculate the new length, why not move it up to the platform layer and just work on floats in the game? You know? Abstracting away the ugly-ness.